test/test_aistore.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+import string
+import tempfile
+import unittest
+
+from torchdata.datapipes.iter import AISFileLister, AISFileLoader
+
+try:
+    from aistore.client.api import Client
+    from aistore.client.errors import AISError, ErrBckNotFound
+
+    AIS_CLUSTER_ENDPT = "http://localhost:8080"
+
+    HAS_AIS = Client(AIS_CLUSTER_ENDPT).is_aistore_running()
+except ImportError or ConnectionError:
+    HAS_AIS = False
+skipIfNoAIS = unittest.skipIf(not HAS_AIS, "AIS not running or library not installed")
+
+
+@skipIfNoAIS
+class TestDataPipeLocalIO(unittest.TestCase):
+    def setUp(self):
+        # initialize client and create new bucket
+        self.client = Client(AIS_CLUSTER_ENDPT)
+        letters = string.ascii_lowercase
+        self.bck_name = "".join(random.choice(letters) for _ in range(10))
+        self.client.create_bucket(self.bck_name)
+        # create temp files
+        num_objs = 10
+
+        # create 10 objects in the `/temp` dir
+        for i in range(num_objs):
+            object_body = "test string" * random.randrange(1, 10)
+            content = object_body.encode("utf-8")
+            obj_name = f"temp/obj{ i }"
+            with tempfile.NamedTemporaryFile() as file:
+                file.write(content)
+                file.flush()
+                self.client.put_object(self.bck_name, obj_name, file.name)
```

Comment:
Would the client copy the tempfile to the bucket? IIRC, `tempfile` will be removed from the fs on the completion of the context.

Reply:
Yes, the file created will be added to the AIS Bucket. Tempfile are deleted as soon as they are closed. In `tearDown()` we are destroying the bucket and all the objects within.